devices/flash-zynq7000/qspi.c

```diff
@@ -269,9 +275,19 @@ static void qspi_IOMode(void)
 
 	/* Configure controller */
 
-	/* Set baud rate to 100 MHz: 200 MHz / 2, master mode, not Legacy mode */
+	/* Set master mode, not Legacy mode */
+	*(qspi_common.base + cr) = 0x1 | (1 << 31);
+
 	*(qspi_common.base + cr) &= ~(0x7 << 3);
-	*(qspi_common.base + cr) |= (0x0 << 3) | 0x1 | (1 << 31);
+	if (QSPI_FCLK != -1) {
+		/* Set baud rate to 100 MHz: 200 MHz / 2 */
+		*(qspi_common.base + cr) |= (0x0 << 3);
```

Comment:
This operation doesn't really make sense. It would be better to avoid writing invalid value to this register in the first place:
```
t = *(qspi_common.base + cr) & ~(0x7 << 3);
if (QSPI_FCLK >= 0) 
  t |= 1 << 3;
*(qspi_common.base + cr) = t;
```

Reply:
Thanks for pointing this out. Fixed.